code/modules/station_goals/shield.dm

```diff
@@ -134,6 +134,32 @@
 	speed_process = TRUE
 	var/kill_range = 14
 
+/obj/machinery/satellite/meteor_shield/sci
+	name = "\improper Meteor Shield Satellite"
+	desc = "A station made meteor point-defense satellite."
+	mode = "M-SHIELD"
+
+/obj/item/disk/meteor
+	name = "Meteor Shield Upgrade Disk"
+	desc = "A floppy disk that allows meteor shields to fire at longer ranges and lowers meteor drawing from gravitational fields.."
+
+/obj/machinery/satellite/meteor_shield/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/disk/meteor))
+		to_chat(user, "<span class='notice'>The disk uploads better tracking and rang modification software.</span>")
+		kill_range = 17
+		change_meteor_chance(0.5)
+	else
+		return ..()
+
+/obj/machinery/satellite/meteor_shield/sci/toggle(user)
+	if(!..(user))
+		return FALSE
+	if(obj_flags & EMAGGED)
+		if(active)
+			change_meteor_chance(8)
+		else
+			change_meteor_chance(1.5)
```

Comment:
This makes it plausible to guarantee meteors by emagging a meteor shield and repeatedly toggling it on/off. Make this 0.125